groups/infrastructure/route-53.tf

```diff
@@ -0,0 +1,21 @@
+resource "aws_acm_certificate" "certificate" {
+  domain_name       = local.domain_name
+  validation_method = "DNS"
+
+  tags = {
+    Environment = var.environment
+  }
+}
+
+resource "aws_route53_record" "route" {
+  name    = aws_acm_certificate.certificate.domain_validation_options[0].resource_record_name
+  type    = aws_acm_certificate.certificate.domain_validation_options[0].resource_record_type
+  zone_id = data.aws_route53_zone.hosted_zone.id
+  records = [aws_acm_certificate.certificate.domain_validation_options[0].resource_record_value]
+  ttl     = 60
+}
```

Comment:
This is legacy syntax and is not valid with the version of the `aws` provider defined in the code. `domain_validation_options` is now exported as a set of objects and so must be iterated over via a `for_each` loop.

The [provider documentation](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/acm_certificate_validation) has a complete example which will also require a small update to the `aws_acm_certificate_validation` resource to fit in with the updated syntax.